sql/src/main/java/io/crate/protocols/postgres/ConnectionContext.java

```diff
@@ -159,32 +161,44 @@
 
     private static final Logger LOGGER = Loggers.getLogger(ConnectionContext.class);
 
-    final MessageDecoder decoder;
-    final MessageHandler handler;
     private final SQLOperations sqlOperations;
     private final Authentication authService;
 
+    private ChannelPipeline pipeline;
+    private SslHandler sslHandler;
     private int msgLength;
     private byte msgType;
     private SQLOperations.Session session;
     private boolean ignoreTillSync = false;
     private SessionContext sessionContext;
 
     enum State {
-        SSL_NEG,
+        PRE_STARTUP,
         STARTUP_HEADER,
         STARTUP_BODY,
         MSG_HEADER,
         MSG_BODY
     }
 
-    private State state = STARTUP_HEADER;
+    private State state = PRE_STARTUP;
 
-    ConnectionContext(SQLOperations sqlOperations, Authentication authService) {
+    private ConnectionContext(Channel channel, SslHandler sslHandler, SQLOperations sqlOperations, Authentication authService) {
+        this.pipeline = channel.pipeline();
```

Comment:
Would it make sense to use access the pipeline on-demand from the ChannelHandlerContext.channel that's available per request in the Decoder? Would remove the Channel dependency to construct the ConnectionContext